docs/real_time_tutorial.md

```diff
@@ -2,101 +2,42 @@
-
-### DDS implementation configuration
-First of all make sure you are running the demo in a RTOS.
-ROS 2 uses DDS as communication middleware. Several DDS implementations are currently supported by ROS 2. In order to achieve real-time capabilities the DDS also has to be configured properly. Each DDS has it's own configuration method, usually based in a xml QoS profile file. Depending on the DDS implementation it is possible to configure the middleware threads priority, lock memory, use static memory allocation, etc.  
-
-We will extend this section in the future providing detailed instructions about how to configure each DDS implementation. (Help is welcome!)
-
-## Examples
-
-For the moment it is not possible in this demo to configure the real-time settings using the ROS 2 launch system. For this reason if we want to tune the real-time settings we have to launch the demo executables using ros2 run.
-
-### Set real-time priority
-
-We can change the process priority and set a real-time FIFO priority. This will set the priority of the thread where the ROS 2 executor is running.
+Example using ros2 launch: 
 
+```bash
+ros2 launch pendulum_bringup pendulum_bringup.launch.py priority:=80 cpu-affinity:=4 lock-memory-size:=100 config-child-threads:=True 
 ```
-ros2 run pendulum_demo pendulum_demo --priority 80
-```
-
-Make sure, we have permissions to change the priority of process in the OS.
-
-### Set CPU affinity
 
-To bind the application to a specific CPU core we can use the option `--cpu-affinity` to set the CPU mask. For example if we want to bind the process to CPU 3 (or 2 starting from 0) we set the CPU mask to 4 (100 in binary).
+Example using the executable command line arguments:
 
-```
-ros2 run pendulum_demo pendulum_demo --priority 80 --cpu-affinity 4
-```
-
-### Lock memory
-
-If our process generates memory page faults when running a real-time task we would suffer an undetermined block time. To prevent this, we must pre-allocate all the dynamic memory used by the process and lock it into RAM.
-
-We can enable memory locking with the option `--lock-memory`.The command will pre-fault memory until no more page faults are seen. This usually allocated a high amount of memory (~8GB) so make sure there is enough memory in the system.
-
-```
-ros2 run pendulum_demo pendulum_demo --lock-memory
-```
-
-Additionally, we can specify the amount of memory we want to pre-allocate with the option `--lock-memory-size`. For example we can pre-allocate 50 MB
-
-```
-ros2 run pendulum_demo pendulum_demo --lock-memory-size 50000
-```
-
-If we get the following error message this means we don't have permissions or that we can't allocate the requested amount of memory.
-
-```
-mlockall failed: Cannot allocate memory
-Couldn't lock  virtual memory.
+```bash
+ros2 run pendulum_demo pendulum_demo --priority 80 --cpu-affinity:=4 --lock-memory-size 100 --config-child-threads True 
```

Comment:
```suggestion
ros2 run pendulum_demo pendulum_demo --priority 80 --cpu-affinity:=4 --lock-memory-size 100 --config-child-threads:=True 
```

Reply:
That only applies to the launch arguments, this is for the executable. Anyhow, we should validate all the commands before approving the PR.